core/src/main/java/tech/tablesaw/columns/Column.java

```diff
@@ -564,6 +564,13 @@ default boolean noneMatch(Predicate<? super T> test) {
 
   Column<T> set(int row, T value);
 
+  default Column<T> set(int row, String newValue, AbstractColumnParser<?> parser) {
+    setObj(row, parser.parse(newValue));
```

Comment:
The reason `appendCell` is implemented on each column type is to avoid auto-boxing. It's maybe a little ugly and I'm not sure how I feel about it, but we should probably do the same here to keep them consistent

Reply:
Sorry, I'm seeing this comment just now. I ended up finding a way to completely remove all `setObj()` implementations at the cost of adding a typecast from `AbstractColumnParser<?>` to `AbstractColumnParser<T>`. Let me know how you feel about this and if it relates to this concern.

Also, can you please elaborate on where auto-boxing happens here?

Reply:
Not sure if you saw my comment on the new version, but the reason the auto-boxing happens is because `parse` returns an Object which must then be converted to a primitive. Every Column has its own implementation of `appendCell` to avoid this. I think we'd want to do the same here for consistency. @lwhite1 implemented this part, so may be able to chime in as well though he's pretty busy at the moment, so we probably shouldn't wait for him

Reply:
Hi @benmccann, can you please take another look? 
I noticed that not all column types use primitive type parsing, so I modified the default implementation to delegate parsing to each specific column implementation which then uses primitive parsing. A default method is still provided to avoid code duplication for column types that don't use primitive types (dates and strings column types).

Reply:
I like the idea of providing a default method so you don't need to implement it on every column. I'd probably just override `set` rather than delegating to a new `setValue` method though to keep the API cleaner for users

Reply:
On second thought, it seems that the missing value check on `setValue()` implementations is not needed, right? It seems that parsers do handle missing values correctly.

Reply:
Yes, that's true actually